Added feature to create file directly

GlassGruber opened the pull request:
Hey there, 
basically I've added a feature to create directly an `.html` file, named like the page title we are archiving; this is enabled with the `-f` flag.
Also I've addressed what I mentioned in #34, and fixed a similar issue with images using `srcset`.

Let me know if there is something valuable!

zTrix commented:
Since this pull request failed the CI test, I've manually merged some part of this pull request.

Thanks again.

GlassGruber commented:
Oh I'm sorry for the build issue, I'm a novice with git in general and CI is something I've not dealt with yet.
What caused the build error specifically?

I noticed you added the output `-o` flag which is convenient; the "file" `-f` flag I've added is instead more of a "lazy mode" switch, basically I don't want to tell every time what file name to use and so on.. 

Also I saw that the `srcset` fix I've added is missing; if you test with current master to get a page with images element using the `srcset` attribute, you will see the browser will do requests for the multiple sources in the `srcset` attribute instead of the `src` that is handled by the script (so if you try to view the page offline, you will get no images at all).

If you wish I can do another PR with the `-f` switch and the `srcset` fix, this time trying to manage the CI build (though, I will need few directions for CI maybe :P)

zTrix commented:
Sorry I've missed the `srcset` part. Will check it later.

For `-f` flag, I noticed the more lazy mode, but didn't figure out a good way to make it more general. Will consider it later.

GlassGruber commented:
Ok, let me know if I can help! 
Just curious, when you say:
> For `-f` flag, I noticed the more lazy mode, but didn't figure out a good way to make it more general.  

what you mean with "more general"?

zTrix commented:
To be compatible with common command line tools like `wget` or `curl`.

GlassGruber commented:
I think of your script more like a stand alone pseudo-curl|wget specifically aimed at archiving an `html` page in a **very** handy single file, a sort of single page hit-man X) 

But yea, I get your point. Let me know if I can help!

zTrix commented:
`srcset` manually merged in latest commit
